server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -745,4 +762,196 @@ private static GetSnapshotsRequestBuilder baseGetSnapshotsRequest(String[] repoN
+                        )
+                    );
+                }
+            }
+        });
+
+        Predicate<SnapshotInfo> filterByNamePredicate = Predicates.always();
+
+        // {repository} path parameter
+        final String[] requestedRepositories;
+        if (randomBoolean()) {
+            requestedRepositories = new String[] { randomFrom("_all", "*") };
+        } else {
+            final var selectedRepositories = Set.copyOf(randomNonEmptySubsetOf(repositories));
+            filterByNamePredicate = filterByNamePredicate.and(si -> selectedRepositories.contains(si.repository()));
+            requestedRepositories = selectedRepositories.toArray(new String[0]);
+        }
+
+        // {snapshot} path parameter
+        final String[] requestedSnapshots;
+        if (randomBoolean()) {
+            requestedSnapshots = randomBoolean() ? Strings.EMPTY_ARRAY : new String[] { randomFrom("_all", "*") };
+        } else {
+            final var selectedSnapshots = randomNonEmptySubsetOf(snapshotInfos).stream()
+                .map(si -> si.snapshotId().getName())
+                .collect(Collectors.toSet());
+            filterByNamePredicate = filterByNamePredicate.and(si -> selectedSnapshots.contains(si.snapshotId().getName()));
+            requestedSnapshots = selectedSnapshots.stream().map(n -> n + "*").toArray(String[]::new);
+        }
+
+        // ?sort and ?order parameters
+        final var sortKey = randomFrom(SnapshotSortKey.values());
+        final var order = randomFrom(SortOrder.values());
+
+        // compute the ordered sequence of snapshots which match the repository/snapshot name filters
+        final var selectedSnapshots = snapshotInfos.stream()
+            .filter(filterByNamePredicate)
+            .sorted(sortKey.getSnapshotInfoComparator(order))
+            .toList();
+
+        final var getSnapshotsRequest = new GetSnapshotsRequest(TEST_REQUEST_TIMEOUT, requestedRepositories, requestedSnapshots)
+            // apply sorting params
+            .sort(sortKey)
+            .order(order);
+
+        // sometimes use ?from_sort_value to skip some items; note that snapshots skipped in this way are subtracted from
+        // GetSnapshotsResponse.totalCount whereas snapshots skipped by ?after and ?offset are not
+        final int skippedByFromSortValue;
+        if (randomBoolean()) {
+            final var startingSnapshot = randomFrom(snapshotInfos);
+            getSnapshotsRequest.fromSortValue(switch (sortKey) {
+                case START_TIME -> Long.toString(startingSnapshot.startTime());
+                case NAME -> startingSnapshot.snapshotId().getName();
+                case DURATION -> Long.toString(startingSnapshot.endTime() - startingSnapshot.startTime());
+                case INDICES, SHARDS -> Integer.toString(startingSnapshot.indices().size());
+                case FAILED_SHARDS -> "0";
```

Comment:
There should be no failed shards. So we effectively do not test it. Should we add a comment?

Reply:
++ see 2f2e4854e5ca08c20a193b332dfd54b000ecfe86.